src/databricks/labs/ucx/install.py

```diff
@@ -619,31 +619,48 @@ def _handle_existing_dashboard(self, dashboard_id: str, display_name: str, paren
             str | None :
                 The dashboard id. If None, the dashboard will be recreated.
         """
-        if "-" in dashboard_id:
-            logger.info(f"Upgrading dashboard to Lakeview: {display_name} ({dashboard_id})")
-            try:
-                self._ws.dashboards.delete(dashboard_id=dashboard_id)
-            except BadRequest:
-                logger.warning(f"Cannot delete dashboard: {display_name} ({dashboard_id})")
-            return None  # Recreate the dashboard if upgrading from Redash
+        if self._is_redash_dashboard(dashboard_id):
+            self._upgrade_redash_dashboard(dashboard_id, display_name)
         try:
             dashboard = self._ws.lakeview.get(dashboard_id)
-            if dashboard.lifecycle_state is None:
-                raise NotFound(f"Dashboard life cycle state: {display_name} ({dashboard_id})")
-            if dashboard.lifecycle_state == LifecycleState.TRASHED:
-                logger.info(f"Recreating trashed dashboard: {display_name} ({dashboard_id})")
+            if self._is_trashed_dashboard(dashboard, display_name, dashboard_id):
                 return None  # Recreate the dashboard if it is trashed (manually)
         except (NotFound, InvalidParameterValue):
-            logger.info(f"Recovering invalid dashboard: {display_name} ({dashboard_id})")
-            try:
-                dashboard_path = f"{parent_path}/{display_name}.lvdash.json"
-                self._ws.workspace.delete(dashboard_path)  # Cannot recreate dashboard if file still exists
-                logger.debug(f"Deleted dangling dashboard {display_name} ({dashboard_id}): {dashboard_path}")
-            except NotFound:
-                pass
-            return None  # Recreate the dashboard if it's reference is corrupted (manually)
+            self._recover_invalid_dashboard(dashboard_id, display_name, parent_path)
+            return None
+        except PermissionDenied:
+            logger.warning(f"Cannot access dashboard {display_name} ({dashboard_id}), permission denied")
+            return None  # Create a new dashboard if permission is denied.
         return dashboard_id  # Update the existing dashboard
 
+    def _is_redash_dashboard(self, dashboard_id: str) -> bool:
+        """Check if the dashboard is a Redash dashboard"""
+        return "-" in dashboard_id
+
+    def _upgrade_redash_dashboard(self, dashboard_id: str, display_name: str) -> None:
+        logger.info(f"Upgrading dashboard to Lakeview: {display_name} ({dashboard_id})")
+        try:
+            self._ws.dashboards.delete(dashboard_id=dashboard_id)
+        except BadRequest:
+            logger.warning(f"Cannot delete dashboard: {display_name} ({dashboard_id})")
+
+    def _is_trashed_dashboard(self, dashboard, display_name: str, dashboard_id: str) -> bool:
+        if dashboard.lifecycle_state is None:
+            raise NotFound(f"Dashboard life cycle state: {display_name} ({dashboard_id})")
```

Comment:
Nit: when raising exceptions, it's a good idea to split the message onto a separate line.
```suggestion
            msg = f"Dashboard life cycle state: {display_name} ({dashboard_id})"
            raise NotFound(msg)
```
Explanation: https://docs.astral.sh/ruff/rules/f-string-in-exception/
(We don't have this rule in our linter, but it's one that I like.)

Reply:
Fixed in https://github.com/databrickslabs/ucx/pull/4210